reactor-core/src/main/java/reactor/core/publisher/ContextHolder.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2011-Present Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.core.publisher;
+
+import reactor.util.context.Context;
+
+interface ContextHolder {
```

Comment:
Why? The lambda usage isn't the primary goal. And even if lambda form is used (e.g. in tests), it will work just fine. From `@FunctionalInterface`'s Javadoc:

> However, the compiler will treat any interface meeting the
> definition of a functional interface as a functional interface
> regardless of whether or not a {@code FunctionalInterface}
> annotation is present on the interface declaration.

Reply:
yeah ok, since the implementing classes are less than likely to be functional interfaces themselves, it doesn't really bring any value.